client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -68,6 +68,8 @@
     
     static final Duration TIME_UNIT = Duration.ofMillis(1000);
     static final Duration UPDATE_WINDOW = Duration.ofMillis(30000);
+    private static final double COMPACTION_PROBABILITY = 0.05;
+    private static final int MIN_UPDATES_BEFORE_COMPACTION = 500;
```

Comment:
Could we do it based on size rather than number of updates? If we compact based on the idea that we don't want to use much space, it would be good to look at size.